bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/MathUtils.java

```diff
@@ -63,4 +67,28 @@ public static long remainder(long numerator, long denominator) {
             }
         }
     }
+
+    public static long addExact(long num1, long num2) {
+        long result = num1 + num2;
+        if (((num1 ^ result) & (num2 ^ result)) < 0) {
+            throwIntRangeOverflowError();
+        }
+        return result;
+    }
+
+    public static long subtractExact(long num1, long num2) {
+        long result = num1 - num2;
+        if (((num1 ^ num2) & (num1 ^ result)) < 0) {
```

Comment:
Shall we add some clarity? Same as above

Reply:
I will add a comment in the code as follows above the if condition
// an overflow has happened on subtraction when both operands have the same sign AND
// if the sign of the result is different from the first operand